core/src/test/scala/kafka/raft/KafkaMetadataLogTest.scala

```diff
@@ -32,13 +32,13 @@ import org.apache.kafka.common.utils.Utils
 import org.apache.kafka.raft.internals.BatchBuilder
 import org.apache.kafka.raft.{KafkaRaftClient, LogAppendInfo, LogOffsetMetadata, OffsetAndEpoch, RecordSerde, ReplicatedLog}
 import org.apache.kafka.snapshot.{SnapshotPath, Snapshots}
-import org.junit.jupiter.api.Assertions.{assertEquals, assertFalse, assertThrows, assertTrue}
+import org.junit.jupiter.api.Assertions.{assertEquals, assertFalse, assertNotEquals, assertThrows, assertTrue}
```

Comment:
This comment applies to all of the tests/changes in this file.

Can we also implement all of the tests that were added to `MockLogTest`? It is good to have tests that validate that this two implementations behave similarly.

Reply:
I add some changes in `MockLog` to delete the elder snapshots, and 2 unit tests.